modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/SortedIndexDescriptor.java

```diff
@@ -17,72 +17,58 @@
 
 package org.apache.ignite.internal.storage.index;
 
-import static java.util.stream.Collectors.toList;
 import static java.util.stream.Collectors.toUnmodifiableList;
 
-import java.util.Arrays;
-import java.util.Comparator;
 import java.util.List;
-import java.util.stream.Stream;
 import org.apache.ignite.configuration.NamedListView;
 import org.apache.ignite.configuration.schemas.table.ColumnView;
 import org.apache.ignite.configuration.schemas.table.IndexColumnView;
 import org.apache.ignite.configuration.schemas.table.SortedIndexView;
 import org.apache.ignite.configuration.schemas.table.TableIndexView;
 import org.apache.ignite.configuration.schemas.table.TableView;
-import org.apache.ignite.internal.schema.Column;
-import org.apache.ignite.internal.schema.SchemaDescriptor;
+import org.apache.ignite.internal.schema.NativeType;
 import org.apache.ignite.internal.schema.configuration.SchemaConfigurationConverter;
 import org.apache.ignite.internal.schema.configuration.SchemaDescriptorConverter;
 import org.apache.ignite.internal.storage.StorageException;
 import org.apache.ignite.internal.tostring.S;
-import org.apache.ignite.schema.definition.ColumnDefinition;
 
 /**
  * Descriptor for creating a Sorted Index Storage.
  *
- * @see SortedIndexStorage
+ * @see SortedIndexMvStorage
  */
 public class SortedIndexDescriptor {
     /**
      * Descriptor of a Sorted Index column (column name and column sort order).
      */
     public static class ColumnDescriptor {
-        private final Column column;
+        private final String name;
 
-        private final boolean asc;
-
-        private final boolean indexedColumn;
+        private final NativeType type;
 
         private final boolean nullable;
 
-        ColumnDescriptor(Column column, boolean asc, boolean indexedColumn, boolean nullable) {
-            this.column = column;
-            this.asc = asc;
-            this.indexedColumn = indexedColumn;
-            this.nullable = nullable;
-        }
+        private final boolean asc;
```

Comment:
Since we're here, you can add "nullFirst" flag, but only if it's easy and won't take much time.

Reply:
I'd rather not, because it's Friday and I'm too tired